cayenne-server/src/test/java/org/apache/cayenne/access/DataContextFlattenedAttributesIT.java

```diff
@@ -233,87 +234,143 @@ public void testSelectColumnQuery() throws Exception {
         ColumnSelect<CompoundPaintingLongNames> originalQuery = ObjectSelect.query(CompoundPaintingLongNames.class)
                 .column(CompoundPaintingLongNames.SELF);
 
-        CompoundPaintingLongNames beforeCompoundPainting = originalQuery.selectFirst(context);
+        CompoundPaintingLongNames beforeCompoundPainting = originalQuery.select(context)
+                .stream()
+                .filter(c -> c.getArtistLongName().equals("artist1"))
+                .findFirst()
+                .orElseThrow();
 
         String beforeArtistNameFromContext = beforeCompoundPainting.getArtistLongName();
         String beforePaintingTitleFromContext = beforeCompoundPainting.getPaintingTitle();
 
-        Object beforeArtistNameFromDatabase = tArtist.selectAll().get(0)[1];
-        Object beforePaintingTitleFromDatabase = tPainting.selectAll().get(0)[1];
+        String beforeArtistNameFromDatabase = tArtist.selectAll().stream()
```

Comment:
Like a comment above, resetting DB state could make this part (and all similar bellow) much cleaner.